[9.0][MIG] medical appointment

lasley opened the pull request:
PR adds the medical_base_history concept & brought over the appointment logic from the base module, using the base_history module for change tracking.

Depends:
- [X] #69 
- [x] #152

lasley commented:
Rebased on top of medical_physician- this PR is now dependent on #86 as well in order to avoid conflicts.

jcdrubay commented on medical_appointment/models/medical_appointment.py:
```diff
@@ -0,0 +1,302 @@
+        help='Check this to ignore any double bookings and schedule anyways',
+    )
+    appointment_date = fields.Datetime(
+        string='Date and Time',
```
I suggest to make this required=True so date_end can be computed. Otherwise I was getting on error message. Also, it seems quite strange to me creating an appointment without even having an hing no the date.

jcdrubay commented on medical_appointment/tests/medical_appointment.yml:
```diff
@@ -0,0 +1,101 @@
+-
```
Some funny characters in this file. ?!

jcdrubay commented on medical_appointment/tests/medical_appointment_stage.yml:
```diff
@@ -0,0 +1,10 @@
+-
```
Funny characters here too.

jcdrubay reviewed (changes requested):
Just few comments after a very quick overview, but this PR is very big... Is it normal? (considering #69 has already been merged)

lasley committed:
[MIG] medical_appointment: Upgrade to v9

lasley commented:
@jcdrubay Alright this has been rebased onto 9.0 and the current incubator has been cross-ported & squashed. Should be easier to review now

lasley commented:
@obulkin - once we have finished on LasLabs#171, let's cherry pick the changes here. I think that is a much better implementation & would prefer to launch OCA 9 on that as well.

obulkin committed:
[10.0] [REM] [IMP] Deprecate medical_base_history and update medical_appointment accordingly (#171)

* [REM] medical_base_history: Remove module
* Remove module, which was only used in medical_appointment and has been
deprecated in favor of auditlog

* [IMP] medical_appointment: Better history tracking
* Fix broken medical.appointment unit test
* Remove all uses of deprecated module medical_base_history from manifest as
well as medical.appointment model and associated data files/views
* Add auditlog dependency and data file with auditlog.rule record for tracking
medical.appointment changes
* Update medical.appointment to inherit from mail.thread for added tracking
and messaging functionality
* Update medical_appointment_view_form to show new mail.thread fields
* Remove comments field from medical.appointment model and associated data
files/views

* PR commit

* [IMP] medical_appointment: Fixes per Review
* Remove DOS line endings from medical_appointment.yml
* Fix formatting of view

lasley commented:
Closing - we no longer have interest in v9. We will continue these forward in v10.

Anyone is welcome to pick the code up and reintroduce a PR, however.
